packages/nx/src/plugins/package-json-workspaces/create-nodes.ts

```diff
@@ -88,22 +88,33 @@ export function buildProjectConfigurationFromPackageJson(
   }
 
   let name = packageJson.name ?? toProjectName(normalizedPath);
-  const projectType =
-    nxJson?.workspaceLayout?.appsDir != nxJson?.workspaceLayout?.libsDir &&
-    nxJson?.workspaceLayout?.appsDir &&
-    directory.startsWith(nxJson.workspaceLayout.appsDir)
-      ? 'application'
-      : 'library';
 
-  return {
+  const projectConfiguration: ProjectConfiguration & { name: string } = {
     root: directory,
     sourceRoot: directory,
     name,
-    projectType,
     ...packageJson.nx,
     targets: readTargetsFromPackageJson(packageJson),
     metadata: getMetadataFromPackageJson(packageJson),
   };
+
+  if (
+    typeof nxJson?.workspaceLayout?.appsDir !== 'undefined' &&
+    typeof nxJson?.workspaceLayout?.libsDir !== 'undefined'
+  ) {
+    projectConfiguration.projectType =
+      nxJson?.workspaceLayout?.appsDir != nxJson?.workspaceLayout?.libsDir &&
+      directory.startsWith(nxJson.workspaceLayout.appsDir)
+        ? 'application'
+        : 'library';
+  } else if (
+    typeof nxJson?.workspaceLayout?.libsDir !== 'undefined' &&
+    directory.startsWith(nxJson.workspaceLayout.libsDir)
+  ) {
+    projectConfiguration.projectType = 'library';
+  }
```

Comment:
How come there's no third case that handles only `appsDir` being set?

Reply:
In that case, we just don't set `projectType`, which was the previous logic. It's either both `appsDir` and `libsDir`, or only `libsDir`. The rest will fall back to `project.type` rather than `project.data.projectType`.